java/kotlin-extractor2/src/main/kotlin/entities/TypeParameter.kt

```diff
@@ -0,0 +1,85 @@
+package com.github.codeql.entities
+
+import com.github.codeql.*
+import com.intellij.psi.PsiElement
+import org.jetbrains.kotlin.analysis.api.KaSession
+import org.jetbrains.kotlin.analysis.api.symbols.KaTypeParameterSymbol
+import org.jetbrains.kotlin.analysis.api.symbols.psiSafe
+import org.jetbrains.kotlin.analysis.api.types.KaTypeParameterType
+import org.jetbrains.kotlin.types.Variance
+
+context(KaSession)
+fun KotlinUsesExtractor.getTypeParameterLabel(param: KaTypeParameterSymbol): String {
+    // Use this instead of `useDeclarationParent` so we can use useFunction with noReplace = true,
+    // ensuring that e.g. a method-scoped type variable declared on kotlin.String.transform<R> gets
+    // a different name to the corresponding java.lang.String.transform<R>, even though
+    // useFunction will usually replace references to one function with the other.
+    val parentLabel = getTypeParameterParentLabel(param)
+    return "@\"typevar;{$parentLabel};${param.name}\""
+}
+
+context(KaSession)
+fun KotlinFileExtractor.extractTypeParameter(
+    tp: KaTypeParameterSymbol,
+    apparentIndex: Int,
+    /* OLD: KE1
+    javaTypeParameter: JavaTypeParameter? */
+): Label<out DbTypevariable>? {
+    with("type parameter", tp) {
+        val parentId = getTypeParameterParentLabel(tp) ?: return null
```

Comment:
Should we log an error in the `null` case?

Reply:
The callee already logs, but I've added error-cascading per this and the comment below.